src/app/catalogue-search/catalogue-search-listing/catalogue-search-listing.component.ts

```diff
@@ -52,6 +54,8 @@ export class CatalogueSearchListingComponent implements OnInit {
     this.parameters = mapStateParamsToSearchParameters(
       this.routerGlobals.params
     );
+    this.parameters.page = this.paginator?.pageIndex ?? 0;
```

Comment:
If redirecting to the page itself when pagination changes as I explained in another comment, then I don't think accessing the `paginator` component would be required, because the query parameters will dictate the page and page size. Suitable defaults will still need to be used in case those query params are missing/undefined though.